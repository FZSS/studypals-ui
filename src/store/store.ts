import { configureStore, Action } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';
import { ThunkAction } from 'redux-thunk';
import lectureReducer from 'store/lecture/slice';
import loginReducer from 'store/login/slice';

const rootReducer = combineReducers({
  lecture: lectureReducer,
  login: loginReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
export type AppThunk = ThunkAction<void, RootState, unknown, Action<string>>;

const store = configureStore({
  reducer: rootReducer,
});

export default store;
