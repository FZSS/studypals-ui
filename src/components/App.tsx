import * as React from 'react';
import 'components/styles.css';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core';
import { accentPurple, primaryPurple, secondaryBlue } from 'utils/styleUtils';
import LoginView from 'components/LoginView/LoginView';
import { useSelector } from 'react-redux';
import { RootState } from 'store/store';
import LectureView from "components/LectureView/LectureView";

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primaryPurple,
      light: accentPurple,
      contrastText: '#ffff',
    },
    secondary: {
      main: secondaryBlue,
      contrastText: '#ffff',
    },
  },
});

const App: React.FunctionComponent = () => {
  const loggedIn = useSelector((state: RootState) => state.login.loggedIn);

  const view = loggedIn ? <LectureView/> : <LoginView/>;
  return (
    <MuiThemeProvider theme={theme}>
      <div className="app-view">{view}</div>
    </MuiThemeProvider>
  );
};

export default App;
