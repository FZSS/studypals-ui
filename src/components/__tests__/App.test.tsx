import * as React from 'react';
import App from '../App';
import * as renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from 'store/store';

test('app matches snapshot', () => {
  const app = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    .toJSON();
  expect(app).toMatchSnapshot();
});
