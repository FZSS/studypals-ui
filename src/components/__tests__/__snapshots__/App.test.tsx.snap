// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app matches snapshot 1`] = `
<div
  className="app-view"
>
  <h4
    className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
  >
    Github Repository List
  </h4>
  <div
    className="repos-header"
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      >
        Keyword
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
      </div>
    </div>
    <div
      className="repos-header-buttons"
    >
      <div
        className="repos-header-button"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Previous
          </span>
        </button>
      </div>
      <div
        className="repos-header-button"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Next
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="repos-table"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      No repositories
    </p>
  </div>
</div>
`;
